src/OpenTelemetry.AutoInstrumentation/Loading/LazyInstrumentationLoader.cs

```diff
@@ -47,34 +50,67 @@ private class OnAssemblyLoadInitializer
         private readonly ILifespanManager _lifespanManager;
         private readonly string _requiredAssemblyName;
 
+#if NETFRAMEWORK
+        private int _firstInitialization = 1;
+#endif
+
         public OnAssemblyLoadInitializer(ILifespanManager lifespanManager, InstrumentationInitializer instrumentationInitializer)
         {
             _instrumentationInitializer = instrumentationInitializer;
             _lifespanManager = lifespanManager;
             _requiredAssemblyName = instrumentationInitializer.RequiredAssemblyName;
+
             AppDomain.CurrentDomain.AssemblyLoad += CurrentDomain_AssemblyLoad;
+
+#if NETFRAMEWORK
+            var isRequiredAssemblyLoaded = AppDomain.CurrentDomain
+                .GetAssemblies()
+                .Any(x => GetAssemblyName(x) == _requiredAssemblyName);
+            if (isRequiredAssemblyLoaded)
+            {
+                OnRequiredAssemblyDetected();
+            }
```

Comment:
I understand that this code is to check if the assembly was already loaded... Maybe it should be not only for .NET Framework? I guess it could happen for .NET as well.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1605#discussion_r1019553855 first bullet.